pkg/controller/customdomain/customdomain_controller.go

```diff
@@ -288,6 +288,19 @@ func (r *ReconcileCustomDomain) Reconcile(request reconcile.Request) (reconcile.
 			return reconcile.Result{}, err
 		}
 	} else {
+		if string(customIngress.Spec.EndpointPublishingStrategy.LoadBalancer.Scope) != ingressScope {
+			errStr := fmt.Sprintf("Invalid update to ingress scope (detected change from %s to %s)", customIngress.Spec.EndpointPublishingStrategy.LoadBalancer.Scope, ingressScope)
+			reqLogger.Info(fmt.Sprintf("The 'scope' field is immutable: detected change from %s to %s. To register a domain with %s scope, a new CustomDomain object will need to be defined.", customIngress.Spec.EndpointPublishingStrategy.LoadBalancer.Scope, ingressScope, ingressScope))
+			SetCustomDomainStatus(
+				reqLogger,
+				instance,
+				errStr,
+				customdomainv1alpha1.CustomDomainConditionInvalidScope,
+				customdomainv1alpha1.CustomDomainStateNotReady)
+			_ = r.statusUpdate(reqLogger, instance)
+			return reconcile.Result{}, errors.New(errStr)
+		}
+
```

Comment:
We should attempt to patch the IngressController here if the scope didn't change. This can be done on a follow-up PR.